tool/planet/etcd.go

```diff
@@ -282,6 +287,91 @@ func etcdUpgrade(rollback bool) error {
 	return nil
 }
 
+// startWatchingEtcdMasters creates a control loop which polls etcd for the etcd cluster member list, and updates the
+// etcd gateway configuration with any changes. This keeps the etcd gateway load balancing in sync with the cluster.
+func startWatchingEtcdMasters(ctx context.Context) error {
+	config := etcdconf.Config{
+		Endpoints: []string{DefaultEtcdEndpoints},
+		KeyFile:   DefaultEtcdctlKeyFile,
+		CertFile:  DefaultEtcdctlCertFile,
+		CAFile:    DefaultEtcdctlCAFile,
+	}
+	cli, err := config.NewClientV3()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	go watchEtcdMasters(ctx, cli)
+	return nil
+}
+
+func watchEtcdMasters(ctx context.Context, client *etcdv3.Client) {
+	ticker := time.NewTicker(15 * time.Second)
+	defer ticker.Stop()
+
+	gateway := etcdGateway{}
```

Comment:
IIUC because this is initialized with an empty `clientURLs`, it will lead to the etcd-gateway restart right away because it will detect the members below and compare them against an empty list.

I would initialize this with current URLs instead. Otherwise, let's say you restart planet-agent and it will in turn trigger etcd-gateway restart.